### PR TITLE
[WASI-NN] ggml: add ErrNo::ModelNotFound

### PR DESCRIPTION
**Pull request by dm4**
To add `ModelNotFound` error to `wasmedge-wasi-nn` and use it in our WasmEdge WASINN examples, we need:

- Merge PR in wasmedge-wasi-nn
  - [PR](https://github.com/second-state/wasmedge-wasi-nn/pull/11)
- Merge PR in WasmEdge runtime
  - [PR](https://github.com/WasmEdge/WasmEdge/pull/3338)
- Release the new wasmedge-wasi-nn crate
- Update the WasmEdge WASINN examples to use the newer version of wasmedge-wasi-nn crate
  - [PR](https://github.com/second-state/WasmEdge-WASINN-examples/pull/130)

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Review by dannypsnl — APPROVED** [review]

File: plugins/wasi_nn/ggml.cpp [hunk 1]
```diff
@@ -709,6 +710,12 @@ Expect<ErrNo> load(WasiNNEnvironment &Env, Span<const Span<uint8_t>> Builders,
     spdlog::info(
         "[WASI-NN][Debug] GGML backend: Finished handling model path."sv);
   }
+  // Check if the model exists.
+  if (!std::filesystem::exists(ModelFilePath)) {
```
Comment (hydai) [inline plugins/wasi_nn/ggml.cpp:714]: Should we handle the utf8 path?

Comment (dm4) [reply plugins/wasi_nn/ggml.cpp:714]: I've tested this on Ubuntu with filenames like `測試.gguf` and works. Should I try some utf8 file path on Windows?

Comment (hydai) [reply plugins/wasi_nn/ggml.cpp:714]: IIRC, there should be a u8path or some classes that can ensure the utf8 path. Could you please check these APIs and make the decision?

**Review by hydai — DISMISSED** [review]

**Review by hydai — APPROVED** [review]

**Pull request merged**